### DISCUSSION TITLE
JavaScriptCore里面持有 js的回调函数 导致循环引用如何解决？

### DISCUSSION THREAD

**Discussion by breakerrorchen** [Tencent/ScriptX#82] [General]
const image = new Image(60, 45); // Using optional size for image
image.onload = function() {
canvas.draw(image)
};
image.src = 'https://mdn.mozillademos.org/files/5397/rhino.jpg';

如果Image是c++实现的，就会出现内存泄漏吧

比如上面的例子 onload 如果被c++层面持有就会有问题。这个用weak 可能回调会失效。如何保证这个场景的正确。

**Comment by LanderlYoung** [comment]
Please show the full code to reproduce.

**Reply by LanderlYoung** [reply]
Also, `onload` is not held by C++. You can read the source code or do some experiment/test.

**Reply by breakerrorchen** [reply]
当然需要被c++持有，不持有下次被GC了，就没有办法调用到了、这就明显的环。在iOS内置的JSC提供的C API接口  几乎没有办法绕过去这个问题。这个可用性就大打折扣了

**Reply by LanderlYoung** [reply]
That is not our implementation, please check out the code. And if there is any bug, please file an ISSUE with full context and code to reproduce.

**Reply by breakerrorchen** [reply]
哈哈 我只是询问，是不是 ScriptX没有办法支持这种场景？在我们看来这是非常常见的场景。我们其实也没有找到办法。
